utf8.h

```diff
@@ -851,6 +858,64 @@ void *utf8codepoint(const void *utf8_restrict str,
   return (void *)s;
 }
 
+size_t utf8codepointsize(int chr) {
+  if (0 == ((int)0xffffff80 & chr)) {
+    return 1;
+  } else if (0 == ((int)0xfffff800 & chr)) {
+    return 2;
+  } else if (0 == ((int)0xffff0000 & chr)) {
+    return 3;
+  } else { // if (0 == ((int)0xffe00000 & chr)) {
+    return 4;
+  }
+}
+
+void *utf8addcodepoint(void *utf8_restrict str, int chr, size_t n) {
```

Comment:
So just to be clear you are assuming that str has been allocated big enough to take the extra codepoint?